etc/kubernetes/manifests/base/apps/cos-fleetshard-sync/observability/service-monitor.yaml

```diff
@@ -0,0 +1,56 @@
+---
+apiVersion: "monitoring.coreos.com/v1"
+kind: "ServiceMonitor"
+metadata:
+  labels:
+    app.kubernetes.io/name: "cos-fleetshard-sync"
+    app.kubernetes.io/part-of: "cos"
+    app.kubernetes.io/version: "latest"
+  name: "cos-fleetshard-sync"
+spec:
+  endpoints:
+    - interval: "30s"
+      path: "/q/metrics"
+      port: "http"
+      scheme: "http"
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: "cos-fleetshard-sync"
+      app.kubernetes.io/part-of: "cos"
```

Comment:
@lgarciaaco for some reason the selector for `ServiceMonitor` is not automatically updated by Quarkus anymore hence I have added a dedicate set of resources for which I have removed the matching of `app.kubernetes.io/version` and it should continue to work as before.

Let me know what you think

Reply:
It looks good to me. The PodMonitors we use for the addon are in https://github.com/bf2fc6cc711aee1a0c2a/cos-tools/tree/main/observability/resources/prometheus/pod_monitors and the label_selector is the name. These are used for deployment don using the templates files generated by Quarkus, provably dev or local environment only.